src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -12,6 +12,11 @@ sealed trait IR extends BaseIR {
 
   override def copy(newChildren: IndexedSeq[BaseIR]): BaseIR =
     Copy(this, newChildren)
+
+  def size: Int = 1 + children.map {
+      case x: IR => x.size
+      case _ => 1
```

Comment:
Maybe irSize and return 0 here?  Or just move this to BaseIR and count everything?  I think for your use case it won't matter since you won't have relational nodes nested in your IRs in Emit.

Reply:
I also have a gut instinct that this should be zero.

Reply:
huh, I totally thought I changed this. I had it as 1 because I dun goofed.